File: adapter/src/main/java/org/jvalue/ods/adapterservice/adapter/model/AdapterConfig.java
package org.jvalue.ods.adapterservice.adapter.model;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.EqualsAndHashCode;

import javax.validation.constraints.NotNull;

@EqualsAndHashCode
public class AdapterConfig {

  @NotNull
  public final ProtocolConfig protocolConfig;

  @NotNull
  public final FormatConfig formatConfig;

  @JsonCreator
  public AdapterConfig(
    @JsonProperty("protocol") ProtocolConfig protocolConfig,
    @JsonProperty("format") FormatConfig formatConfig) {
    this.protocolConfig = protocolConfig;
    this.formatConfig = formatConfig;
  }
}

File: adapter/src/model/AdapterConfig.ts
import { FormatConfig } from "./FormatConfig";
import { ProtocolConfig } from "./ProtocolConfig";

export class AdapterConfig {
  protocolConfig: ProtocolConfig;
  formatConfig: FormatConfig;
  //TODO JSON CREATER FROM JAVA
  /*
  @JsonCreator
  public AdapterConfig(
    @JsonProperty("protocol") ProtocolConfig protocolConfig,
    @JsonProperty("format") FormatConfig formatConfig) {
    this.protocolConfig = protocolConfig;
    this.formatConfig = formatConfig;
  }
  */
  constructor(protocolConfig: ProtocolConfig, formatConfig: FormatConfig) {
    this.protocolConfig = protocolConfig;
    this.formatConfig = formatConfig;
  }
}
